lib/matplotlib/backends/backend_svg.py

```diff
@@ -590,7 +596,7 @@ def _get_clip_attrs(self, gc):
         clippath, clippath_trans = gc.get_clip_path()
         if clippath is not None:
             clippath_trans = self._make_flip_transform(clippath_trans)
-            dictkey = (id(clippath), str(clippath_trans))
+            dictkey = (self._get_next_clip_id(), str(clippath_trans))
```

Comment:
I think this breaks the re-use of a clip path if we have multiple artists that use the same clip path.

We need to keep a second dictionary that maps `id(clippath) -> incrementing int` so that on line 608 when we make the oid we can use that instead.

Reply:
It wont let me comment on line 608, but I think the logic there should be something like

```python
if clip is None:
    stable_id = self._get_next_clip_id()
    self._dictkey_to_id[dictkkey] = stable_id
    oid = self._make_id('p', stable_id)
```

Reply:
Ok, yep.  It seems like some extra test coverage would be useful in that case, to confirm whether keeping an id-mapping is required (and to test any fixes if so).

Reply:
I'm not sure the test coverage I've added makes semantic/usage sense, but it does now cover sharing a clip-path (technically a `Patch`) object across multiple artists.

I've added some test coverage on ID-uniqueness in the generated SVG at the same time after finding some code/issue history mentioning that it's important to maintain distinct identifiers.

Reply:
(however, as noted in #27831 - I missed a note that recommended a particular style of test coverage - it's not yet added here)

Reply:
> so I don't think we actually need a second dictionary to keep the mapping? That seems good to not pick up extra state.

The implication of this is that further refactoring in this part of the code could be helpful -- and I agree that there seem to be opportunities to simplify the logic here.  However: I think it's important to confirm that the test coverage is sufficient first.